### PR TITLE
19-Extract-and-label-data-based-on-user-input

### PR DESCRIPTION
**Pull request by 1AhmedYasser**

### CODE REVIEW

File: GUI/src/services/entities.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import api from './api';
+import { Entity } from '../types/entity';
+
+export async function addEntity(entityData: { name: string }) {
+  const { data } = await api.post('entities', entityData);
+  console.log(`a7a ${data}`)
```
Comment (baha-a) [inline GUI/src/services/entities.ts:6]: Remove the console.log statement.

File: DSL/OpenSearch/init.sh [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
+#intents
+curl -H "Content-Type: application/x-ndjson" -X PUT "http://localhost:9200/intents" -ku admin:admin --data-binary "@fieldMappings/intents.json"
+curl -H "Content-Type: application/x-ndjson" -X PUT "http://localhost:9200/intents/_bulk" -ku admin:admin --data-binary "@mock/intents.json"
+curl -L -X POST 'http://localhost:9200/_scripts/search_intents' -H 'Content-Type: application/json' --data-binary "@templates/search_intents.json"
+
 #services
 curl -H "Content-Type: application/x-ndjson" -X PUT "http://localhost:9200/services" -ku admin:admin --data-binary "@fieldMappings/services.json"
 curl -H "Content-Type: application/x-ndjson" -X PUT "http://localhost:9200/services/_bulk" -ku admin:admin --data-binary "@mock/services.json"
-curl -L -X POST 'http://localhost:9200/_scripts/get-services-stat' -H 'Content-Type: application/json' -H 'Cookie: customJwtCookie=test' --data-binary "@templates/get-services-stat.json"
+curl -L -X POST 'http://localhost:9200/_scripts/get-services-stat' -H 'Content-Type: application/json' -H 'Cookie: customJwtCookie=test' --data-binary "@templates/get-services-stat.json"
```
Comment (baha-a) [inline DSL/OpenSearch/init.sh:12]: Add empty line here.

File: DSL/OpenSearch/fieldMappings/intents.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        "fielddata": true,
+        "fields": {
+          "raw": {
+            "type": "keyword"
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (baha-a) [inline DSL/OpenSearch/fieldMappings/intents.json:24]: Add empty line here.

File: DSL/Opensearch/fieldMappings/services.json [hunk 4]
```diff
@@ -9,4 +9,4 @@
       }
     }
   }
-}
+}
```
Comment (baha-a) [inline DSL/Opensearch/fieldMappings/services.json:12]: Add empty line here.

File: GUI/src/pages/Training/IntentsFollowupTraining/index.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        onValueChange={setSelectedTab}
+      >
+        <Tabs.List
+          className='vertical-tabs__list'
+          aria-label={t('training.intents.followupTraining') || ''}
+        >
+          {[
+            { label: t('training.intents.entities'), value: 'entities' },
+            { label: t('training.intents.regex'), value: 'regex' },
+          ].map((tab, index) => (
```
Comment (baha-a) [inline GUI/src/pages/Training/IntentsFollowupTraining/index.tsx:38]: Please refactor this array into a variable for better readability.

**Review by baha-a — CHANGES_REQUESTED** [review]
If you're using VS Code you could configure it to add empty line at the end of each file automatically on save.

File: GUI/src/utils/generateUEID.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export const generateUEID = () => {
```
Comment (trevorling) [inline GUI/src/utils/generateUEID.ts:1]: I'd prefer to use the `UUID` library instead of a custom solution.

**Review by trevorling — CHANGES_REQUESTED** [review]

File: GUI/src/pages/Training/IntentsFollowupTraining/index.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  useEffect(() => {
+    const queryTab = searchParams.get('tab');
+    if (queryTab) {
+      setSelectedTab(queryTab);
+    }
+  }, [searchParams]);
+
+  return (
+    <>
+      <h1>{t('training.intents.followupTraining')}</h1>
```
Comment (Minwasko) [inline GUI/src/pages/Training/IntentsFollowupTraining/index.tsx:23]: It looks like this translation is missing from the i18n file. Please add all missing entries there.

**Review by Minwasko — CHANGES_REQUESTED** [review]

**Comment by trevorling** [comment]
@1AhmedYasser 
Please resolve conflicts

**Review by trevorling — APPROVED** [review]
@1AhmedYasser 
Please go over the previous comments- related with `let` and `const`.

**Review by Minwasko — APPROVED** [review]

**Review by baha-a — APPROVED** [review]

**Review by trevorling — APPROVED** [review]

**Pull request merged**